### PR TITLE
cid gravity integration

### PR DESCRIPTION
**Pull request by anjor**

### CODE REVIEW

**Review by Lusitaniae — APPROVED** [review]

File: dataprep-tools/filecoin/boost_create_deals.py [hunk 1]
```diff
@@ -81,6 +85,90 @@
...
+
+    resp = response.json()
+    assert (
+        "result" in resp
+    ), f"result key not found in response from CID gravity: {resp}"
+    assert (
+        "providers" in resp["result"]
+    ), f"providers key not found in response from CID gravity: {resp}"
+
+    return resp["result"]["providers"]
```
Comment (ze42) [inline dataprep-tools/filecoin/boost_create_deals.py:169]: Might be nice to check if `providers` is empty, and log `reason`.

**Review by ze42 — COMMENTED** [review]
Also some checks currently done would most likely be useless later on.

* You should never get the `skipping xxx - already have deal with X` if you send the deal directly
* You should not have the `already replicated X times` if you have the same copy count configured in CIDGravity

For now keep those checks as you already have them, and let us know if you still get those messages, so we could try to investigate together.

File: dataprep-tools/filecoin/boost_create_deals.py [hunk 2]
```diff
@@ -81,6 +85,90 @@
...
+    )
+    assert response.status_code == 200, f"failed to get chain head: {response.text}"
+
+    resp = response.json()
+    assert "result" in resp, f"result key not found in ChainHead response: {resp}"
+    assert (
+        "Height" in resp["result"]
+    ), f"Height key not found in ChainHead response: {resp}"
+
+    return resp["result"]["Height"]
```
[the remark this answers is not in the document]
Comment (anjor) [reply dataprep-tools/filecoin/boost_create_deals.py:102]: I prefer getting it from the chain.

File: dataprep-tools/filecoin/boost_create_deals.py [hunk 3]
```diff
@@ -81,6 +85,90 @@
...
+    padded_size,
+):
+
+    head = get_chain_head()
+    start_epoch = head + start_epoch_head_offset
+
+    provider_collateral = get_collateral(padded_size=padded_size, verified=verified)
+
+    api_key = environ.get("CID_GRAVITY_KEY")
+    headers = {"X-API=KEY": api_key}
```
[the remark this answers is not in the document]
Comment (anjor) [reply dataprep-tools/filecoin/boost_create_deals.py]: nice catch, thanks

File: dataprep-tools/filecoin/boost_create_deals.py (hunk 1 above)
Comment (anjor) [reply dataprep-tools/filecoin/boost_create_deals.py:174]: done

**Pull request merged**